### DISCUSSION TITLE
Choose sourcesmap library for mapped_code util

### DISCUSSION THREAD

**Discussion by linrium** [pintariching/rustle#16] [Ideas]
I checked this file https://github.com/sveltejs/svelte/blob/master/src/compiler/utils/mapped_code.ts. 
As I see in this file, it imports three libraries.

```js
import { DecodedSourceMap, RawSourceMap, SourceMapLoader } from '@ampproject/remapping/dist/types/types';
import remapping from '@ampproject/remapping';
import { SourceMap } from 'magic-string';
```

We can use https://github.com/h-a-n-a/magic-string-rs for `magic-string`. With @ampproject/remapping we can use https://github.com/getsentry/rust-sourcemap or https://www.npmjs.com/package/@swc-node/sourcemap-support.

I haven't checked it carefully. Feel free to help.

**Comment by pintariching** [comment]
I've added `magic-string-rs` as a dependency. It seems well maintained. For @ampproject/remapping I would like to use a `swc` crate, but I'm still looking into it.
